### PR TITLE
doc: how to change default rbd object size

### PR DESCRIPTION
**Pull request by LittleFox94**
This pull request adds a hint to the documentation how to configure the default rbd object size and improves the documentation for the `rbd_default_order` config value (adding it to `rbd-config-ref` and improving it's docs).

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by LittleFox94** [comment]
Hm some tests are failing, I guess that one change in `rbd.yaml.in` wasn't a doc-only change after all ..

File: doc/man/8/rbd.rst [hunk 1]
```diff
@@ -81,6 +81,9 @@ Parameters
    nearest power of two; if no suffix is given, unit B is assumed.  The default
    object size is 4M, smallest is 4K and maximum is 32M.
 
+   The default value can be changed with the config ``rbd_default_order``, which
```
Comment (idryomov) [inline doc/man/8/rbd.rst:84]:
```suggestion
   The default value can be changed with the configuration option ``rbd_default_order``, which
```

File: doc/man/8/rbd.rst [hunk 2]
```diff
@@ -81,6 +81,9 @@ Parameters
...
+   takes a power to two (default object size is ``2 ^ rbd_default_order``).
```
Comment (idryomov) [inline doc/man/8/rbd.rst:85]:
```suggestion
   takes a power of two (default object size is ``2 ^ rbd_default_order``).
```

File: src/common/options/rbd.yaml.in [hunk 3]
```diff
@@ -481,6 +481,10 @@ options:
   type: uint
   level: advanced
   desc: default order (data block object size) for new images
+  long_desc: This configures the default object size for new rbd images. The value is used as
```
Comment (idryomov) [inline src/common/options/rbd.yaml.in:484]:
```suggestion
  long_desc: This configures the default object size for new images. The value is used as
```

File: src/common/options/rbd.yaml.in [hunk 4]
```diff
@@ -481,6 +481,10 @@ options:
...
+    power to two, meaning ``default_object_size = 2 ^ rbd_default_order`` (alternative
```
Comment (idryomov) [inline src/common/options/rbd.yaml.in:485]:
```suggestion
    a power of two, meaning ``default_object_size = 2 ^ rbd_default_order``
```

File: src/common/options/rbd.yaml.in [hunk 5]
```diff
@@ -481,6 +481,10 @@ options:
...
+    format ``default_object_size = 1 << rbd_default_order``). Configure a value between 12 and 25
```
Comment (idryomov) [inline src/common/options/rbd.yaml.in:486]: The `1 << ` syntax is fairly niche, most people are confused by it.  Let's drop it.
```suggestion
    Configure a value between 12 and 25
```

**Review by idryomov — CHANGES_REQUESTED** [review]
Just a couple a of nits and please squash everything into a single commit.

**Comment by LittleFox94** [comment]
Thanks for looking at it @idryomov, changed everything you commented except squashing into a single commit, since I read the [contribution guidelines](https://github.com/ceph/ceph/blob/master/doc/start/documenting-ceph.rst#commit-the-change) as if I should make one commit per file (though the second commit is touching two files already, having multiple commits started from that though)? I will happily squash them, but want to make sure with the docs first :)

File: src/common/options/rbd.yaml.in [hunk 6]
```diff
@@ -481,6 +481,9 @@ options:
   type: uint
   level: advanced
   desc: default order (data block object size) for new images
+  long_desc: This configures the default object size for new images. The value is used as
+    power to two, meaning ``default_object_size = 2 ^ rbd_default_order``. Configure a value
```
Comment (idryomov) [inline src/common/options/rbd.yaml.in:485]:
```suggestion
    a power of two, meaning ``default_object_size = 2 ^ rbd_default_order``. Configure a value
```

**Review by idryomov — CHANGES_REQUESTED** [review]
Otherwise LGTM.

These commits are tiny and closely related to each other, no need to split per-file or otherwise.

File: src/common/options/rbd.yaml.in (hunk 6 above)
Comment (LittleFox94) [reply src/common/options/rbd.yaml.in]: whoops, missed that

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test api

**Pull request merged**